c7n/resources/elasticsearch.py

```diff
@@ -116,6 +116,58 @@ class KmsFilter(KmsRelatedFilter):
     RelatedIdsExpression = 'EncryptionAtRestOptions.KmsKeyId'
 
 
+@ElasticSearchDomain.filter_registry.register('cross-cluster-search-connections')
+class ElasticSearchSearchConnections(ValueFilter):
+    """Check for Cloudfront distribution config values
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: elasticsearch-cross-cluster-search-connections
+            resource: aws.elasticsearch
+            filters:
+              - type: cross-cluster-search-connections
+                key: outbound[].SourceDomainInfo.OwnerId
+                value: 0123456789012
+                op: contains
+   """
+
+    schema = type_schema(
+        'cross-cluster-search-connections', rinherit=ValueFilter.schema,)
+    schema_alias = False
+    annotation_key = "c7n:SearchConnections"
+    annotate = False
+    permissions = ('es:ESCrossClusterGet',)
+
+    def process(self, resources, event=None):
+        client = local_session(self.manager.session_factory).client('es')
+        results = []
+
+        for r in resources:
+            try:
+                inbound = self.manager.retry(
```

Comment:
fetching the connections should only be done if the annotation key doesn't already exist